// node-pty/index.js
'use strict';
// Platform/libc dispatcher for the vendored node-pty. macOS + Windows use the
// mainline node-pty prebuilds (N-API); glibc Linux uses @lydell (N-API); musl
// Linux uses @homebridge (per-Node-ABI). Each sub-dir keeps its own lib next to
// its own binary, so there is no cross-fork native-API mismatch.
//
// glibc + mac/win are N-API, so they load on any Node. The musl source is
// per-Node-ABI, so it only loads when a prebuild matching the running Node's
// ABI was vendored; assertMuslPrebuildPresent() turns a missing ABI into a
// clear, actionable error instead of @homebridge's cryptic from-source
// 'Cannot find module ../build/Release/pty.node'.
function isMuslLinux() {
  if (process.platform !== 'linux') return false;
  const report = typeof process.report?.getReport === 'function' ? process.report.getReport() : undefined;
  const header = report && typeof report === 'object' ? report.header : undefined;
  const glibcVersionRuntime = header && typeof header === 'object' ? header.glibcVersionRuntime : undefined;
  // glibc runtimes report a version string here; musl does not.
  return !(typeof glibcVersionRuntime === 'string' && glibcVersionRuntime.length > 0);
}
function assertMuslPrebuildPresent() {
  const fs = require('fs');
  const path = require('path');
  const abi = process.versions.modules;
  const prebuildsDir = path.join(__dirname, 'musl', 'prebuilds', 'linux-' + process.arch);
  const expected = path.join(prebuildsDir, 'node.abi' + abi + '.musl.node');
  if (fs.existsSync(expected)) return;
  let availableAbis = [];
  try {
    availableAbis = fs
      .readdirSync(prebuildsDir)
      .map(function (name) {
        const matched = /^node\.abi(\d+)\.musl\.node$/.exec(name);
        return matched ? matched[1] : undefined;
      })
      .filter(Boolean)
      .sort(function (a, b) {
        return Number(a) - Number(b);
      });
  } catch (readError) {
    // Fall through with an empty list; the ABI mismatch is the real error.
  }
  throw new Error(
    'node-pty (vendored): no musl/Alpine prebuild for Node ABI ' +
      abi +
      ' (' +
      process.version +
      ', ' +
      process.arch +
      '). Bundled musl ABIs: [' +
      availableAbis.join(', ') +
      ']. Run on a glibc Linux image (any Node version), or on Alpine/musl use a Node version whose ABI is listed above.',
  );
}
let impl;
if (process.platform === 'linux' && isMuslLinux()) {
  assertMuslPrebuildPresent();
  impl = require('./musl/lib/index.js');
} else if (process.platform === 'linux') {
  impl = require('./glibc/lib/index.js');
} else {
  impl = require('./mac-win/lib/index.js');
}
// Re-export node-pty's named bindings as literal exports.<name> = assignments so
// Node's ESM loader detects them; module.exports = require(<dynamic target>)
// exposes only default, leaving "import * as pty" -> pty.spawn undefined.
Object.defineProperty(exports, '__esModule', { value: true });
exports.spawn = impl.spawn;
exports.fork = impl.fork;
exports.createTerminal = impl.createTerminal;
exports.open = impl.open;
exports.native = impl.native;
